packages/block-library/package.json

```diff
@@ -45,6 +45,7 @@
 		"@wordpress/date": "file:../date",
 		"@wordpress/deprecated": "file:../deprecated",
 		"@wordpress/dom": "file:../dom",
+		"@wordpress/editor": "file:../editor",
```

Comment:
👋🏻 There are a bunch of warnings suggesting that @wordpress/block-library should not depend on @wordpress/editor as blocks can be loaded into a *non-post* block editor. 

In other places, to avoid declaring @wordpress/editor as a dependency, the store is accessed  using the key string, e.g., `select( 'core/editor' )`.

Example: https://github.com/WordPress/gutenberg/blob/d1a8aaa54a0cc65bc06fc1d242919afcf4dc4df8/packages/server-side-render/src/index.js#L18-L18

Not sure how else to get the post's current id (aside from the URL params that is) 🤔

Reply:
Thanks for pointing this out! Having looked at some other instances where the key string was used I think it should be okay to do that here too.

I've added fallbacks to ensure this works in the site editor, or if the store isn't available at all.

Reply:
No, we shouldn't do that, even with a hardcoded string. We've tried to cleanup any remaining similar implementations and we should definitely not add more. Can you share the other instances you found? We should see to remove them.

The approach in such cases is to see how we can pass down the information needed. For this case I think the `postId` from block's context is already there.